conans/test/unittests/tools/build/test_cppstd.py

```diff
@@ -67,6 +67,8 @@ def test_supported_cppstd_gcc(compiler, compiler_version, values):
     ("apple-clang", "9.5", ['98', 'gnu98', '11', 'gnu11', "14", "gnu14", "17", "gnu17"]),
     ("apple-clang", "10", ['98', 'gnu98', '11', 'gnu11', "14", "gnu14", "17", "gnu17", "20",
                            "gnu20"]),
+    ("apple-clang", "13", ['98', 'gnu98', '11', 'gnu11', "14", "gnu14", "17", "gnu17", "20",
+                           "gnu20", "23", "gnu23"]),
```

Comment:
```suggestion
    ("apple-clang", "13", ['98', 'gnu98', '11', 'gnu11', "14", "gnu14", "17", "gnu17", "20"]),
```

Reply:
This looks like it will fix the test but is this the real issue?

Reply:
but is c++23 supported by apple-clang 13? I could not see mentions in the man page

Reply:
clang++ 13.0 accepts c++2b and gnu++2b.

Reply:
> but is c++23 supported by apple-clang 13? I could not see mentions in the man page

https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes

> New C++ projects you create in Xcode use C++20 language dialect by default. (93456065)
Several C++20 and C++2b papers have been implemented:
C++20 papers that have been implemented:
P0692R1 - Access checking on specializations
P0388R4 - Permit conversions to arrays of unknown bound
C++2b papers that have been implemented:
P1938R3 - if consteval
P1401R5 - Narrowing contextual conversions to bool
P1949R7 - C++ identifier syntax using Unicode Standard Annex 31
P2360R0 - Extend init statement to allow alias declaration (93898598)

Reply:
> New C++ projects you create in Xcode use C++20 language dialect by default. (93456065)

This means that Xcode sets some project property to c++20, so it calls clang with that. clang itself without --std= still defaults to gnu++98. I've checked this by comparing output of `clang++ -x c++ -dM -E /dev/null` with output of `clang++ -x c++ -dM --std=c++98 -E /dev/null` and `clang++ -x c++ -dM --std=gnu++98 -E /dev/null`. I've also checked __cplusplus value in a compiled file.

Manual pages are wrong; they are most probably taken from vanilla clang.

I've checked list of supported --std= values by running `clang++ --std=123 test.cpp`:
* clang++ from Xcode 12.5.1 (12.0.5/clang-1205.0.22.11) **does not** support c++2b/gnu++2b
* clang++ from Xcode 13.0 (13.0.0/clang-1300.0.29.3) **does** support c++2b/gnu++2b

Reply:
Apple Clang supports already a fair amount of C++2b features and I'm looking to enabled them in our projects as soon as we get the Conan support :)

https://en.cppreference.com/w/cpp/23

Reply:
I've updated the 2.x code so the test passes.

Note that there is duplicate information in `conan/tools/_compilers.py` and `conan/tools/build/cppstd.py`. I think the former could be implemented using the latter (something like check the flag for all known standards, and return all tha have flag != None).